### PR TITLE
Sync CI with conda

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Changes:
- remove replay
- switch hardcoded GitHub Action commits in favor of tags
- sync with https://github.com/conda/conda/pull/13162
- sync with https://github.com/conda/conda/pull/13497

### Checklist - did you ...

- [ ] ~Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [ ] ~Add / update necessary tests?~
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -181,77 +184,75 @@ jobs:
...
         if: '!cancelled()'
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          path: ${{ env.REPLAY_DIR }}
-          name: ${{ env.REPLAY_NAME }}-${{ matrix.test-type }}
+          name: test-results-${{ env.HASH }}
+          path: |
+            .coverage
+            tools\durations\${{ runner.os }}.json
```
Comment (jezdez) [inline .github/workflows/tests.yml:253]: We don't have durations yet

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -279,74 +281,72 @@ jobs:
...
         if: '!cancelled()'
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: ${{ env.REPLAY_NAME }}-${{ matrix.test-type  }}
-          path: ${{ env.REPLAY_DIR }}
+          name: test-results-${{ env.HASH }}
+          path: |
+            .coverage
+            tools/durations/${{ runner.os }}.json
```
Comment (jezdez) [inline .github/workflows/tests.yml:347]: Same

**Review by jezdez — DISMISSED** [review]
Shall we add the release artifact upload as well?

**Comment by jezdez** [comment]
Seems like the tests are legitimately broken on Windows with 3.8/3.12?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (kenodegard) [reply .github/workflows/tests.yml]: See https://github.com/conda/conda-build/pull/5146/commits/600d868b1d436497d7b823b3e8d9a29c71023a66

File: .github/workflows/tests.yml (hunk 2 above)
Comment (kenodegard) [reply .github/workflows/tests.yml]: See https://github.com/conda/conda-build/pull/5146/commits/600d868b1d436497d7b823b3e8d9a29c71023a66

**Review by jezdez — APPROVED** [review]

**Pull request merged**